app/queries/get_change_offer_options.rb

```diff
@@ -0,0 +1,84 @@
+class GetChangeOfferOptions
+  attr_accessor :user, :for_provider, :recruitment_cycle_year
+
+  # application_choice.offered_course (either current or the original)
+  def initialize(user:, current_course:)
+    @user = user
+    @for_provider = current_course.accredited_provider || current_course.provider
+    @recruitment_cycle_year = current_course.recruitment_cycle_year
+  end
+
+  def make_decisions_courses
```

Comment:
I'm a bit confused as to why we have two different methods retrieving courses. Within the context of `GetChangeOfferOption` don't we just need one? Also can the one we need to use be called `courses` instead?

Reply:
The one you need for the wizard is `offerable_courses`, but you'll probably restrict it to the chosen provider, e.g. `service.offerable_courses.where(provider: ...)`

The `current_course:` argument to the service determines the ratifying provider to perserve. In most cases, this will be `application_choice.offered_course`, hence the comment we'll remove.

We could rename `offerable_courses` to `courses`, but isn't this a bit less descriptive? You won't be just calling `.courses` in the wizard.

As for the existence `make_decisions_courses`, we're trying to isolate this part of the query in order to test it properly, as it's a database-level re-implementation of `can_make_decisions?` from `ProviderAuthorisation`. Once we establish this works correctly in the rspec examples, we can stop worrying about permissions and test the other stuff (like preserving the ratifying body) without worrying about which complicated permissions setup we're bundling into the other tests.

The `make_decisions_courses` aspect could eventually be moved to a different part of the code, such as `User` or `ProviderAuthorisation`.

Reply:
Maybe it would make more sense then to move this method in it's own service, so it's descriptive enough and can be tested properly. Then it should also be reusable if required by other services/code that is not part of the make offer funcitonality.


> We could rename offerable_courses to courses, but isn't this a bit less descriptive? You won't be just calling .courses in the wizard.

We would be calling `GetChangeOfferOption.new(...).courses` which I think is sufficiently descriptive.